### PR TITLE
changing unset array index into null

### LINKED ISSUE #15640: Trying to access array offset on value of type null [Results.php#4476]
### Describe the bug

Trying to access array offset on value o type null in file Display\Results.php

### To Reproduce

1. Go to 'Empty created table'

### Expected behavior

In older versions of PHP this will not break code (just brings exception) but in 7.4 it breaks.

### Screenshots

1. https://prnt.sc/q98dwf
image

2. https://prnt.sc/q988sa
image

### Server configuration

- Operating system: Ubuntu 16.04
- Web server: Apache
- PHP version: 7.4
- phpMyAdmin version: 5.1.0-dev, latest stable version: 4.9.2
- Server: Localhost via UNIX socket
- Server type: MariaDB
- Server connection: SSL is not being used Documentation
- Server version: 10.0.38-MariaDB-0ubuntu0.16.04.1 - Ubuntu 16.04
- Protocol version: 10
- Server charset: UTF-8 Unicode (utf8mb4)

### Client configuration

- Browser: Chrome
- Operating system: Windows

### Additional context

Possible fix:
1. https://prnt.sc/q98j6h
2. https://prnt.sc/q98jog

### PR DESCRIPTION
**Pull request by rahgurung**
closes #15640 
Signed-off-by: R. Gurung <gurungrahul2@gmail.com>

### CODE REVIEW

File: libraries/classes/Display/Results.php [hunk 1]
```diff
@@ -1223,8 +1223,11 @@ private function _getTableHeadersForColumns(
                 // Results can't be sorted
                 $html
                     .= $this->_getDraggableClassForNonSortableColumns(
-                        $col_visib, $col_visib_current, $condition_field,
-                        $fields_meta[$i], $comments
+                        $col_visib,
+                        isset($col_visib[$j]) ? $col_visib[$j] : null,
```
Comment (williamdes) [inline libraries/classes/Display/Results.php]: Can you remove the (maybe) unused variable on line 1186 that mentions php 7.4 and has a similar fix please?

**Review by williamdes — APPROVED** [review]

File: libraries/classes/Display/Results.php (hunk 1 above)
Comment (gurrrung) [reply libraries/classes/Display/Results.php]: I tried but it gives the same error.

**Comment by williamdes** [comment]
@gurrrung I made the change to group the 2 use cases 
can you test it or give me a way to test ?
The issue did not give enough info to reproduce it

**Comment by gurrrung** [comment]
> @gurrrung I made the change to group the 2 use cases
> can you test it or give me a way to test ?
> The issue did not give enough info to reproduce it

I usually created a table with no data and view it or explore around already created tables to see if any error comes up.

File: libraries/classes/Display/Results.php [hunk 2]
```diff
@@ -1184,7 +1184,7 @@ private function _getTableHeadersForColumns(
 
         for ($j = 0; $j < $number_of_columns; $j++) {
             // PHP 7.4 fix for accessing array offset on bool
-            $col_visib_current = is_array($col_visib) ? $col_visib[$j] : null;
```
Comment (gurrrung) [inline libraries/classes/Display/Results.php]: Yes, this doesn't gives me any error.

**Review by rahgurung — COMMENTED** [review]
LGTM

File: libraries/classes/Display/Results.php [hunk 3]
```diff
@@ -1184,7 +1184,7 @@ private function _getTableHeadersForColumns(
...
+            $col_visib_current = is_array($col_visib) && isset($col_visib[$j]) ? $col_visib[$j] : null;
```
Comment (gurrrung) [inline libraries/classes/Display/Results.php]: Yes, this doesn't gives me any error.

**Comment by williamdes** [comment]
Thank you for your contribution @gurrrung !

**Pull request merged**